bench-tps/src/bench.rs

```diff
@@ -466,21 +488,43 @@ fn do_tx_transfers<T: BenchTpsClient>(
             let transfer_start = Instant::now();
             let mut old_transactions = false;
             let mut transactions = Vec::<_>::new();
+            let mut min_timestamp = u64::MAX;
             for tx in txs0 {
                 let now = timestamp();
                 // Transactions that are too old will be rejected by the cluster Don't bother
                 // sending them.
+                if tx.1 < min_timestamp {
+                    min_timestamp = tx.1;
+                }
                 if now > tx.1 && now - tx.1 > 1000 * MAX_TX_QUEUE_AGE {
                     old_transactions = true;
                     continue;
                 }
                 transactions.push(tx.0);
             }
 
+            if min_timestamp != u64::MAX {
+                datapoint_info!(
+                    "bench-tps-do_tx_transfers",
+                    ("oldest-blockhash-age", timestamp() - min_timestamp, i64),
+                );
```

Comment:
Double checking for my understanding here. This should be a relatively small number since all transactions in this queue should have been updated pretty recently in the call to `generate_txs` https://github.com/solana-labs/solana/blob/b705b51fab73757fef6aa204a1ae891dfd5df57c/bench-tps/src/bench.rs#L396 right?

Reply:
Yes, when things are working as they should, this should be a relatively small value. However, I wanted to add this stat to verify that that was the case, in case for example, the previous send took a long time.

Reply:
unless there's some really stale transactions in the queue, but seems like those should have all been popped off here: https://github.com/solana-labs/solana/blob/b705b51fab73757fef6aa204a1ae891dfd5df57c/bench-tps/src/bench.rs#L481-L484

Reply:
Yeah, this was really added as a way of verifying that that was the case and debugging, in case for example, the send of the previous batch took an unexpectedly long time for whatever reason (e.g. we spent lots of time in get_connection due to lock contention).